android/engine/src/main/java/org/smartregister/fhircore/engine/util/PractitionerDetailsUtils.kt

```diff
@@ -0,0 +1,159 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.util
+
+import com.google.android.fhir.FhirEngine
+import com.google.gson.Gson
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.hl7.fhir.r4.model.CareTeam
+import org.hl7.fhir.r4.model.ListResource
+import org.hl7.fhir.r4.model.Location
+import org.hl7.fhir.r4.model.Organization
+import org.hl7.fhir.r4.model.Parameters
+import org.hl7.fhir.r4.model.Resource
+import org.hl7.fhir.r4.model.ResourceType
+import org.hl7.fhir.r4.model.StringType
+import org.smartregister.fhircore.engine.data.remote.model.response.UserInfo
+import org.smartregister.fhircore.engine.util.extension.asReference
+import org.smartregister.fhircore.engine.util.extension.decodeResourceFromString
+import org.smartregister.fhircore.engine.util.extension.encodeJson
+import org.smartregister.fhircore.engine.util.extension.encodeResourceToString
+import org.smartregister.model.practitioner.KeycloakUserDetails
+import org.smartregister.model.practitioner.PractitionerDetails
+
+@Singleton
+class PractitionerDetailsUtils
+@Inject
+constructor(val sharedPreferences: SharedPreferencesHelper, val fhirEngine: FhirEngine) {
+
+  suspend fun getResourceFromPractitionerDetails(resourceType: String): List<Resource> {
+    val practitionerDetails: Parameters =
+      sharedPreferences.read(PRACTITIONER_PARAMETERS_SHARED_PREFERENCE_KEY, "")!!
+        .decodeResourceFromString()
+    return when (resourceType) {
+      ResourceType.CareTeam.name ->
+        return getResourcesList(
+          practitionerDetails,
+          ResourceType.CareTeam.name,
+          CareTeam::class.java
+        )
+      ResourceType.Organization.name ->
+        return getResourcesList(
+          practitionerDetails,
+          ResourceType.Organization.name,
+          Organization::class.java
+        )
+      ResourceType.Location.name ->
+        return getResourcesList(
+          practitionerDetails,
+          ResourceType.Location.name,
+          Location::class.java
+        )
+      else -> arrayListOf()
+    }
+  }
+
+  suspend fun <T : Resource> getResourcesList(
+    practitionerDetails: Parameters,
+    resourceName: String,
+    clazz: Class<T>
+  ): ArrayList<T> {
+    val resourceList = arrayListOf<T>()
+    practitionerDetails.parameter.forEach {
+      if (it.name.equals(resourceName)) {
+        val result = it.resource as ListResource
+        if (result.hasEntry()) {
+          result.entry.forEach { entry ->
+            resourceList.add(
+              fhirEngine.load(
+                clazz = clazz,
+                id = entry.item.reference.replace("$resourceName/", "")
+              )
+            )
+          }
+        }
+      }
+    }
+    return resourceList
+  }
```

Comment:
We can consider using kotlin lambdas to improve readability
```suggestion
    return practitionerDetails
      .parameter
      .filter { parameterComponent ->
        parameterComponent.hasName() &&
          parameterComponent.name.equals(resourceName, ignoreCase = true) &&
          parameterComponent.resource is ListResource
      }
      .flatMap { (it.resource as ListResource).entry }
      .map { entryComponent ->
        fhirEngine.load(
          clazz = clazz,
          id = entryComponent.item.reference.replace("$resourceName/", "")
        )
      }
```